pkg/config/updater.go

```diff
@@ -0,0 +1,49 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-errors/errors"
+	v1API "github.com/supabase/cli/pkg/api"
+)
+
+type ConfigUpdater struct {
+	client v1API.ClientWithResponses
+}
+
+func NewConfigUpdater(client v1API.ClientWithResponses) ConfigUpdater {
+	return ConfigUpdater{client: client}
+}
+
+func (u *ConfigUpdater) UpdateRemoteConfig(ctx context.Context, remote baseConfig) error {
+	if err := u.UpdateApiConfig(ctx, remote.ProjectId, remote.Api); err != nil {
+		return err
+	}
+	// TODO: implement other service configs, ie. auth
+	return nil
+}
+
+func (u *ConfigUpdater) UpdateApiConfig(ctx context.Context, projectRef string, c api) error {
```

Comment:
I've moved most of your [implementation](https://github.com/supabase/branching/pull/163/files#diff-2f6ae6efa668b8891389d9d65b8b18cefc13b580c4a4c16138c75b26d9df8cbcR13) here because it can be reused for `config push` in the future.

Reply:
Right ! Plus it allow to get rid of the public `RemoteApi` I needed in branching.

I wonder if the "functions upsert" we have on branching already should fit there as well, since it's kindda in-between a config and something else.

Reply:
I see what you mean. The function bundling part is very tricky because we have to convert the config to docker paths when bundling from cli, but leave them as host paths when ran from branching executor.

May be we can unify the approaches if edge runtime starts releasing cross platform binaries which allow would allow us to only use the OS native path.